frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx

```diff
@@ -127,13 +129,6 @@ const useServiceDeployment = () => {
       return false;
     }
 
-    const masterSafeOlasBalance = masterSafeBalances?.find(
-      (walletBalanceResult) =>
-        walletBalanceResult.symbol === TokenSymbol.OLAS &&
-        walletBalanceResult.evmChainId === selectedAgentConfig.evmHomeChainId,
-    )?.balance;
-
-    const isInitialFunded = storeState?.[selectedAgentType]?.isInitialFunded;
     if (service && isInitialFunded && isServiceStaked) {
```

Comment:
@truemiller can you please look at this code and the `" // SERVICE IS STAKED, AND STARTING AGAIN"` code section below and tell me if I'm right or not - it looks, that if a user runs the agent for the second time, the code will alway end up here, and no "threshold check" check will happen (to check if a user has enough gas for starting the agent)

also, here we check this: `(serviceTotalStakedOlas ?? 0) >= requiredStakedOlas` but if a user staked already, their total staked will be more than required anyway, what sense does this make?